csrc/dynamic_transform.h

```diff
@@ -23,15 +23,43 @@ class Fusion;
 class ExpressionEvaluator;
 class DynamicTransformInfoBuilder;
 
+//! Compute the IterType of an IterDomain that has been resized. If the output
+//! is size 1, or the output uses no input elements, this function returns
+//! Broadcast. Otherwise, it returns Iteration.
+inline IterType resize_output_itertype(
+    int64_t in_extent,
+    int64_t out_extent,
+    int64_t left,
+    int64_t right) {
+  TORCH_CHECK(out_extent >= 0, "Resized extent must be non-negative.");
+  if (
+      // negative padding sums to input extent. Output is zero-dimensional
+      out_extent == 0 ||
+      // input overlaps output
+      left + in_extent > 0 || right + in_extent > 0) {
```

Comment:
You're right, this is wrong and needlessly complicated. We assert `out_extent >= 0` above this so as it's currently written I think the else branch is never taken. I think we should just set itertype to `Broadcast` iff `out_extent == 0`.

PyTorch throws an error whenever left or right is less than `-in_extent`, even if the other pad would compensate to result in a positive `out_extent`. I think that's probably pretty reasonable. 

Going with the PyTorch behavior, we could still pure pad value output; for example `left, right = -in_extent, n` for some `n`. It's a very rare corner case, but we could replace it with `full(pad_value)`. I doubt it's worth the trouble to do the `full` replacement, since folks would probably never do that intentionally.

Reply:
Do we need to handle these corner cases in an expr-by-expr basis? For example, what about `slice` with positive and negative factors? 

For now, I think it's okay to incrementally add more complete support of corner cases. Let's just make it clear where these conditions should be handled for each particular expression.